app/src/main/java/com/example/sw0b_001/Models/GatewayClients/GatewayClientViewModel.kt

```diff
@@ -20,17 +26,32 @@ class GatewayClientViewModel : ViewModel() {
         return liveData
     }
 
+    fun updateSelectedGatewayClient(newGatewayClientMsisdn: String?) {
+        val newGatewayClient = liveData.value?.find { it.mSISDN == newGatewayClientMsisdn }
+        _selectedGatewayClient.value = newGatewayClient
+    }
+
     fun loadRemote(context: Context,
                    successRunnable: Runnable?,
                    failureRunnable: Runnable?){
         CoroutineScope(Dispatchers.Default).launch{
             try {
                 GatewayClientsCommunications.fetchAndPopulateWithDefault(context)
+
+                val gatewayClient = GatewayClientsCommunications(context)
+                val defaultMsisdn = gatewayClient.getDefaultGatewayClient()
+                val defaultGatewayClient = defaultMsisdn?.let { getGatewayClientByMsisdn(context, it) }
+
+                _selectedGatewayClient.value = defaultGatewayClient
                 successRunnable?.run()
             } catch (e: Exception) {
                 Log.e(javaClass.name, "Exception fetching Gateway clients", e)
                 failureRunnable?.run()
             }
         }
     }
+
+    private fun getGatewayClientByMsisdn(context: Context, msisdn: String): GatewayClient? {
```

Comment:
This is good, look for the function below in GatewayCLientListingFragment and use the sharedpreference change to fetch from getGatewayClientByMsisdn and make the necessary changes 

``` 
private val sharedPreferencesChangeListener = OnSharedPreferenceChangeListener { _, _ ->
        if(::listViewAdapter.isInitialized)
            listViewAdapter.notifyDataSetChanged()
    }
   ```